Add Example Files

gomesjason opened the pull request.

hughdanliu commented on examples/kubernetes/dynamic-provisioning/README.md:
```diff
@@ -0,0 +1,55 @@
+
+## Context
+
+FSx OpenZFS resources can be mounted to a give pod either statically or dynamically.
```
FSx for OpenZFS* given*

hughdanliu commented on examples/kubernetes/dynamic-provisioning/README.md:
```diff
@@ -0,0 +1,55 @@
+Static provisioning requires the user to initially create the FSxZ resource manually.
+Dynamic provisioning will automatically create FSxZ resources based on user specifications.
+
+This guide will detail steps needed to dynamically create and mount a FSxZ resource.
```
an* FSxZ resource, this applies elsewhere

hughdanliu commented on examples/kubernetes/dynamic-provisioning/README.md:
```diff
@@ -0,0 +1,55 @@
+For details on statically mounting a FSxZ resource see this [guide](../static-provisioning/README.md).
+
+This CSI driver supports the use of both FSxZ filesystems and FSxZ volumes as a container storage interfaces.
+This guide will detail steps needed to deploy both types of resources.
```
detail the steps*

hughdanliu commented on examples/kubernetes/dynamic-provisioning/README.md:
```diff
@@ -0,0 +1,55 @@
+
+## Usage
+
+This example shows you how to create, and mount a FSxZ filesystem and FSxZ volume dynamically.
```
unnecessary comma, I think it reads better if we just say "This example shows you how to dynamically provision an FSxZ file system and FSxZ volume in your cluster" (or something along those lines)

hughdanliu commented on examples/kubernetes/dynamic-provisioning/README.md:
```diff
@@ -0,0 +1,55 @@
+
+### Create and Mount a FSxZ Resource
+
+1. Run **one** following to apply manifests needed to create either a FSxZ filesystem or a FSxZ volume.
```
one of the following*
to apply the manifests necessary*

hughdanliu commented on examples/kubernetes/dynamic-provisioning/README.md:
````diff
@@ -0,0 +1,55 @@
+1. Run **one** following to apply manifests needed to create either a FSxZ filesystem or a FSxZ volume.
+    1. FSxZ filesystem:
+    ```sh
+   kubectl apply -f filesystem/,manifests/
````
Nice! I like this notation, didn't know you could use commas to separate them out like this.

hughdanliu commented on examples/kubernetes/dynamic-provisioning/filesystem/storageclass.yaml:
```diff
@@ -0,0 +1,28 @@
+  dailyAutomaticBackupStartTime: "00:00"
+  deploymentType: "SINGLE_AZ_1"
+  diskIopsConfiguration: "Iops=300,Mode=USER_PROVISIONED"
+  rootVolumeConfiguration: "CopyTagsToSnapshots=true,DataCompressionType=NONE,NfsExports=[{ClientConfigurations=[{Clients=*,Options=[rw,crossmnt]}]}],ReadOnly=false,RecordSizeKiB=128,UserAndGroupQuotas=[{Id=1,StorageCapacityQuotaGiB=10,Type=USER}]"
```
Can you add comments indicating which parameters are required?

hughdanliu commented on examples/kubernetes/dynamic-provisioning/volume/storageclass.yaml:
```diff
@@ -0,0 +1,22 @@
+  copyTagsToSnapshots: "false"
+  dataCompressionType: "NONE"
+  nfsExports: "[{ClientConfigurations=[{Clients=*,Options=[rw,crossmnt]}]}]"
+  parentVolumeId: "fsvol-03062e7ff37662dff"
```
same here, add comments for volumeType and parentVolumeId to indicate that they're required

hughdanliu commented on examples/kubernetes/snapshot/README.md:
```diff
@@ -0,0 +1,97 @@
+
+## Context
+
+Volume snapshots allows users to quickly store the state of volume at a given time.
```
Volume snapshots allow*
state of a* volume

hughdanliu commented on examples/kubernetes/snapshot/README.md:
```diff
@@ -0,0 +1,97 @@
+
+Currently only FSxZ volume level snapshots are supported. 
+FSxZ filesystem level backups are NOT supported.
+See the [design document](../../../docs/design.md) for more information.
```
The design document doesn't exist. Do you plan on adding one in the future? I'm not convinced it's necessary.

gomesjason replied:
I planned on adding a design doc along with a parameters doc. The design doc would point out specific decisions and limitations such as throughput not being expandable, and why SkipFinalBackups is required. I'm not opposed to doing a FAQ instead although.

hughdanliu commented on examples/kubernetes/snapshot/README.md:
```diff
@@ -0,0 +1,97 @@
+## Context
+
+Volume snapshots allows users to quickly store the state of volume at a given time.
+Snapshots can be taken either manually, or utilizing the CSI driver.
```
either manually or by utilizing the CSI driver*

hughdanliu commented on examples/kubernetes/snapshot/README.md:
```diff
@@ -0,0 +1,97 @@
+
+Volume snapshots allows users to quickly store the state of volume at a given time.
+Snapshots can be taken either manually, or utilizing the CSI driver.
+Any snapshot taken can be utilized to create new volumes.
```
I'd add that we can only create a new volume from a snapshot if it exists under the same file system. That is a current limitation of our ZFS snap-send-rcv feature.

hughdanliu commented on examples/kubernetes/snapshot/README.md:
```diff
@@ -0,0 +1,97 @@
+This example shows you how to create snapshots and restore them.
+
+Values in the example files may be modified or removed based on preferences.
+See this [document](../../../docs/parameters.md) for parameter configurations.
```
This file doesn't exist yet, but we should add it eventually. Were you planning on adding it later?

gomesjason replied:
Yea I plan on adding it later when I do the documents task.

hughdanliu commented on examples/kubernetes/snapshot/README.md:
````diff
@@ -0,0 +1,97 @@
+    ```
+
+### Create VolumeSnapshot From PersistentVolume
+1. Run **one** following to apply manifests needed to create either a FSxZ filesystem or a FSxZ volume.
````
one of the following*

hughdanliu commented on examples/kubernetes/snapshot/snapshot/class.yaml:
```diff
@@ -0,0 +1,8 @@
+kind: VolumeSnapshotClass
+metadata:
+  name: fsx-snapshot-class
+driver: fsx.openzfs.csi.aws
```
fsx.openzfs.csi.aws.com*

gomesjason replied:
Good catch. You saved me some headache when testing this morning.

hughdanliu commented on examples/kubernetes/snapshot/existing-snapshot/content.yaml:
```diff
@@ -0,0 +1,13 @@
+  source:
+    snapshotHandle: fsvolsnap-0809d8f1df40a7427
+  driver: fsx.openzfs.csi.aws.com
+  deletionPolicy: Delete
```
nit: extra lines at the end of files

hughdanliu commented on examples/kubernetes/static-provisioning/README.md:
```diff
@@ -0,0 +1,59 @@
+
+## Context
+
+FSxZ resources can be mounted to a give pod either statically or dynamically.
```
Not going to dive super deep into this one, but most of the comments left on the README for dynamic provisioning are also applicable here.

hughdanliu commented on examples/kubernetes/static-provisioning/volume/pv.yaml:
```diff
@@ -0,0 +1,22 @@
+    - timeo=600
+  persistentVolumeReclaimPolicy: Delete
+  csi:
+    driver: fsx.openzfs.csi.aws
```
fsx.openzfs.csi.aws.com*

hughdanliu commented on examples/kubernetes/volume-expansion/README.md:
```diff
@@ -0,0 +1,50 @@
+Users are able to change the capacity of their volumes. 
+Currently, the capacity can ONLY be increased, and NOT decreased.
+Only storage capacity can be adjusted. Throughput can NOT be modified.
+See the [design document](../../../docs/design.md) for more information.
```
no design document yet, also I'd add a disclaimer that scaling the capacity on an FSxZ volume increases the StorageCapacityQuota and StorageCapacityReservation to the same value (the requested storage)

hughdanliu commented on examples/kubernetes/dynamic-provisioning/README.md:
```diff
@@ -0,0 +1,55 @@
+This guide will detail steps needed to dynamically create and mount a FSxZ resource.
+For details on statically mounting a FSxZ resource see this [guide](../static-provisioning/README.md).
+
+This CSI driver supports the use of both FSxZ filesystems and FSxZ volumes as a container storage interfaces.
```
as container storage interfaces*

gomesjason committed:
Add example files

Adds example files on how to static provision, dynamic provision, expand volume capacity, and utilize snapshots
